src/arrow_parquet/uri_utils.rs

```diff
@@ -139,6 +194,78 @@ async fn get_s3_object_store(bucket_name: &str) -> AmazonS3 {
     aws_s3_builder.build().unwrap_or_else(|e| panic!("{}", e))
 }
 
+async fn get_azure_object_store(container_name: &str) -> MicrosoftAzure {
+    let mut azure_builder = MicrosoftAzureBuilder::new().with_container_name(container_name);
+
+    if is_testing() {
+        // use azurite for testing
+        azure_builder =
+            azure_builder.with_endpoint("http://localhost:10000/devstoreaccount1".into());
+        azure_builder = azure_builder.with_allow_http(true);
+    }
+
+    // ~/.azure/config
+    let azure_config_file_path = std::env::var("AZURE_CONFIG_FILE").unwrap_or(
```

Comment:
why not use from_env()?

it's a bit surprising that we're inflicting the environment variables on ourselves

Reply:
for some of the tests, we need to make sure only a subset of the all env vars should exist to verify that different authors methods or scenarios work.